hw3/01-characters.js

```diff
@@ -1,4 +1,75 @@
-// url for the Thrones API
-const url = 'https://thronesapi.com/api/v2/Characters';
+const url = "https://thronesapi.com/api/v2/Characters";
 
-fetch(url);
+let list = document.querySelector("#list");
+
+const fetchData = (url) => {
+  return new Promise(async (resolve, reject) => {
+    try {
+      const response = await fetch(url);
+      const data = await response.json();
+      resolve(data);
+    } catch (error) {
+      reject(error);
+    }
+  });
+};
```

Comment:
I believe this can be simplified by making `fetchData` an `async` function: 
```js
const fetchData = async (url) => ...
```

Then `resolve(data)` can be replaced with `return data` and `reject(error)` can be replaced with `throw error`.